tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp

```diff
@@ -0,0 +1,595 @@
+  const std::array<double, 3> random_point{
+      {radius * sin(theta) * cos(phi) + center[0],
+       radius * sin(theta) * sin(phi) + center[1],
+       radius * cos(theta) + center[2]}};
+
+  const auto test_helper = [&random_point, &time, &functions_of_time](
+                               const auto& map_to_test) noexcept {
+    test_serialization(map_to_test);
+    CHECK_FALSE(map_to_test != map_to_test);
+
+    test_coordinate_map_argument_types(map_to_test, random_point, time,
+                                       functions_of_time);
+    test_jacobian(map_to_test, random_point, time, functions_of_time);
+    test_inv_jacobian(map_to_test, random_point, time, functions_of_time);
+    test_inverse_map(map_to_test, random_point, time, functions_of_time);
+  };
+  test_helper(map);
+
+  const auto check_map_equality = [&random_point, &time, &functions_of_time](
+                                      const auto& map_one,
+                                      const auto& map_two) {
+    tnsr::I<double, 3, Frame::Logical> source_point{};
+    for (size_t i = 0; i < 3; ++i) {
+      source_point.get(i) = gsl::at(random_point, i);
+    }
+    CHECK_ITERABLE_APPROX(map_one(source_point, time, functions_of_time),
+                          map_two(source_point, time, functions_of_time));
+    CHECK_ITERABLE_APPROX(
+        map_one.jacobian(source_point, time, functions_of_time),
+        map_two.jacobian(source_point, time, functions_of_time));
+    CHECK_ITERABLE_APPROX(
+        map_one.inv_jacobian(source_point, time, functions_of_time),
+        map_two.inv_jacobian(source_point, time, functions_of_time));
+  };
+  const auto map2 = serialize_and_deserialize(map);
+  check_map_equality(
+      domain::make_coordinate_map<Frame::Logical, Frame::Grid>(map),
+      domain::make_coordinate_map<Frame::Logical, Frame::Grid>(map2));
+  test_helper(map2);
+}
+
+template <bool InteriorMap>
+void test_map(gsl::not_null<std::mt19937*> generator) noexcept {
+  INFO("Map");
+  std::uniform_real_distribution<> phi_dis(0, 2.0 * M_PI);
+  std::uniform_real_distribution<> theta_dis(0, M_PI);
+
+  // Create a map, choose a time, and create a FunctionOfTime.
+  // Set up the map
+  CoordinateMaps::TimeDependent::SphericalCompression<InteriorMap> map{};
+  double time{std::numeric_limits<double>::signaling_NaN()};
+  std::unordered_map<std::string,
+                     std::unique_ptr<::domain::FunctionsOfTime::FunctionOfTime>>
+      functions_of_time{};
+
+  double min_radius{std::numeric_limits<double>::signaling_NaN()};
+  double max_radius{std::numeric_limits<double>::signaling_NaN()};
+  std::array<double, 3> center{std::numeric_limits<double>::signaling_NaN()};
+  generate_map_time_and_f_of_time(
+      make_not_null(&map), make_not_null(&time),
+      make_not_null(&functions_of_time), make_not_null(&min_radius),
+      make_not_null(&max_radius), make_not_null(&center), generator);
+
+  const double theta{theta_dis(*generator)};
```

Comment:
[optional] this code used to create a random point could be factored into another small helper function. It's used three separate times across the tests.